File: unittest/scripts/py_devapi/scripts/dba_cluster_rpl_user_password.py
# Assumptions: smart deployment routines available

#@ Initialization
deployed_here = reset_or_deploy_sandboxes()

shell.connect({'host': localhost, 'port': __mysql_sandbox_port1, 'user': 'root', 'password': 'root'})

# Install validate_password plugin and configure it for the medium policy
installed = False
try:
  session.run_sql('INSTALL PLUGIN validate_password SONAME \'' + __plugin + '\'');
  installed = True
except:
  pass #This means the plugin is already installed

session.run_sql('SET GLOBAL validate_password_policy=\'MEDIUM\'');

#@ Create cluster
if __have_ssl:
  cluster = dba.create_cluster('pmCluster', {'memberSslMode': 'REQUIRED'})
else:
  cluster = dba.create_cluster('pmCluster')

# configure the validate_password plugin for the strong policy
session.run_sql('SET GLOBAL validate_password_policy=\'STRONG\'');

#@ Add instance 2 to cluster
add_instance_to_cluster(cluster, __mysql_sandbox_port2)

session.run_sql('SET GLOBAL validate_password_length=33');

#@ Add instance 3 to cluster
add_instance_to_cluster(cluster, __mysql_sandbox_port3)

if installed:
	session.run_sql('UNINSTALL PLUGIN validate_password');

session.close();

#@ Finalization
if deployed_here:
    cleanup_sandboxes(deployed_here)

File: unittest/scripts/js_devapi/scripts/dba_cluster_rpl_user_password.js
// Assumptions: smart deployment routines available

//@ Initialization
testutil.deploySandbox(__mysql_sandbox_port1, "root", {report_host: hostname});
testutil.deploySandbox(__mysql_sandbox_port2, "root", {report_host: hostname});
testutil.deploySandbox(__mysql_sandbox_port3, "root", {report_host: hostname});

shell.connect(__sandbox_uri1);

// Install validate_password plugin and configure it for the medium policy
var installed = false;
try {
  session.runSql('INSTALL PLUGIN validate_password SONAME /*(*/\'' + __plugin + '\'/*)*/');
  installed = true;
} catch (err) {
  // This means the plugin is already installed
}

session.runSql('SET GLOBAL validate_password_policy=\'MEDIUM\'');

//@ Create cluster
if (__have_ssl)
  var cluster = dba.createCluster('devCluster', {memberSslMode: 'REQUIRED'});
else
  var cluster = dba.createCluster('devCluster', {memberSslMode: 'DISABLED'});

// configure the validate_password plugin for the strong policy
session.runSql('SET GLOBAL validate_password_policy=\'STRONG\'');

//@ Add instance 2 to cluster
testutil.waitMemberState(__mysql_sandbox_port1, "ONLINE");
cluster.addInstance(__sandbox_uri2);

// configure the validate_password plugin validate_password_length to maximum allowed
session.runSql('SET GLOBAL validate_password_length=32');

//@ Add instance 3 to cluster
cluster.addInstance(__sandbox_uri3);

if (installed)
	session.runSql('UNINSTALL PLUGIN validate_password');

session.close();
cluster.disconnect();

//@ Finalization
testutil.destroySandbox(__mysql_sandbox_port1);
testutil.destroySandbox(__mysql_sandbox_port2);
testutil.destroySandbox(__mysql_sandbox_port3);
